### PR TITLE
feat: support k8s and kubesphere version without "v"

### PR DESCRIPTION
**Pull request by zaunist**
### What type of PR is this?
/kind feature

### What this PR does / why we need it:
This makes kubekey more convenient to use and improves the user experience
### Which issue(s) this PR fixes:

Fixes https://github.com/kubesphere/kubekey/issues/1365

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
Now, we can use both "1.23.6" and "v1.23.6" to create clusters.
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/common/loader.go [hunk 1]
```diff
@@ -109,7 +109,11 @@ func (d *DefaultLoader) Load() (*kubekeyapiv1alpha2.Cluster, error) {
 		Registry: {hostname},
 	}
 	if d.KubernetesVersion != "" {
-		s := strings.Split(d.KubernetesVersion, "-")
+		ver := normalizedBuildVersion(d.KubernetesVersion)
+		if len(ver) < 2 {
```
Comment (mangoGoForward) [inline pkg/common/loader.go:113]: Use regex to check the version more better, like https://github.com/kubernetes/kubernetes/blob/89aaf7c02dd33415ac994cecfddca7c02ed75834/cmd/kubeadm/app/util/version.go#L153

File: pkg/common/loader.go [hunk 2]
```diff
@@ -127,7 +131,11 @@ func (d *DefaultLoader) Load() (*kubekeyapiv1alpha2.Cluster, error) {
 	}
 
 	if d.KubeSphereEnable {
-		if err := defaultKSConfig(&allInOne.Spec.KubeSphere, d.KubeSphereVersion); err != nil {
+		ver := normalizedBuildVersion(d.KubeSphereVersion)
+		if len(ver) < 2 {
```
Comment (mangoGoForward) [inline pkg/common/loader.go:135]: Same as above

**Review by mangoGoForward — COMMENTED** [review]
Nice work, thanks your contribution.
And please confirm whether the file of `go.sum` needs to be modified

File: pkg/common/loader.go (hunk 1 above)
Comment (zaunist) [reply pkg/common/loader.go]: Thanks your advice. :)

**Comment by mangoGoForward** [comment]
Hi @zaunist , seems this place also need to process https://github.com/kubesphere/kubekey/blob/7c761db76c5aed62289fda073e43c3c8283e9879/pkg/common/loader.go#L232-L246

**Review by mangoGoForward — COMMENTED** [review]
Good to me
/LGTM

**Comment by zaunist** [comment]
@pixiake PTAL.

File: pkg/common/loader.go [hunk 3]
```diff
@@ -230,18 +239,22 @@ func (f FileLoader) Load() (*kubekeyapiv1alpha2.Cluster, error) {
...
+			return nil, errors.New(fmt.Sprintf("Unsupported Kubesphere Version: %v\n", f.KubeSphereVersion))
+		}
+		if err := defaultKSConfig(&clusterCfg.Spec.KubeSphere, ver); err != nil {
 			return nil, err
 		}
 	}
 
-	if f.KubernetesVersion != "" {
-		s := strings.Split(f.KubernetesVersion, "-")
+	if ver := normalizedBuildVersion(f.KubernetesVersion); ver != "" {
```
Comment (24sama) [inline pkg/common/loader.go:251]: `f.KubernetesVersion` is generated by cmd's flag:
https://github.com/kubesphere/kubekey/blob/7c761db76c5aed62289fda073e43c3c8283e9879/cmd/ctl/create/cluster.go#L133

So, I guess normalizing this variable may not work for the Kubernetes version specified in the config file.

What do you think about normalizing `clusterCfg.Spec.Kubernetes.Version` directly?

**Comment by 24sama** [comment]
Hi @zaunist , thanks for your contributions!

**Comment by zaunist** [comment]
> Hi @zaunist , thanks for your contributions!

Good catch, thanks for your carefulness :)

**Comment by 24sama** [comment]
LGTM, the last thing, could you please squash your commits @zaunist ? The git history will look more clear.

**Comment by 24sama** [comment]
Thanks @zaunist 

/approve

**Comment by pixiake** [comment]
@all-contributors please add @zaunist for code

**Comment by pixiake** [comment]
@all-contributors please add @mangoGoForward  for review

**Pull request merged**